packages/untp-test-suite/integration/cli/untpTest.integration.test.ts

```diff
@@ -0,0 +1,228 @@
+    });
+  });
+
+  describe('process test runner and final report return WARN', () => {
+    let credentialFileName: string;
+    let storePath: string;
+    let stdout: any;
+    let credentials: ConfigContent[];
+
+    const mockPath = `${process.cwd()}/integration/mock/untpTestWarn`;
+    beforeAll((done) => {
+      credentialFileName = 'credentialsExample.json';
+      storePath = `${mockPath}/${credentialFileName}`;
+      const fileContent = fs.readFileSync(storePath, 'utf8');
+
+      try {
+        credentials = JSON.parse(fileContent).credentials;
+      } catch (error) {
+        expect(error).toBeNull();
+      }
+
+      exec(`yarn untp test -c ${storePath}`, (error, result) => {
+        if (error) {
+          console.error(`execSync error: ${error}`);
+          return;
+        }
+
+        stdout = result;
+        done();
+      });
+    });
+
+    it('should exist the volume in quantityList of objectEvent data', () => {
+      const data = fs.readFileSync(`${process.cwd()}/${credentials[0].dataPath}`, 'utf8');
+      const jsonData = JSON.parse(data);
+      expect(jsonData.quantityList[0].volume).toEqual(expect.any(String));
+    });
+
+    describe('generate test suite result by template', () => {
+      it('should show WARN in the report when data invalidate', () => {
+        expect(stdout).toMatch(/WARN/);
+        expect(stdout).toMatch(/Your credentials are UNTP compliant, but have extended the data model/);
+      });
+    });
+  });
+
+  describe('process test runner with result combine FAIL and WARN and final report return FAIL', () => {
+    let credentialFileName: string;
+    let storePath: string;
+    let stdout: any;
+    let credentials: ConfigContent[];
+
+    const mockPath = `${process.cwd()}/integration/mock/untpTestFailuresAndWarnings`;
+    beforeAll((done) => {
+      credentialFileName = 'credentialsExample.json';
+      storePath = `${mockPath}/${credentialFileName}`;
+      const fileContent = fs.readFileSync(storePath, 'utf8');
+
+      try {
+        credentials = JSON.parse(fileContent).credentials;
+      } catch (error) {
+        expect(error).toBeNull();
+      }
+
+      exec(`yarn untp test -c ${storePath}`, (error, result) => {
+        if (error) {
+          console.error(`execSync error: ${error}`);
+          return;
+        }
+
+        stdout = result;
+        done();
+      });
+    });
+
+    it('should exist the volume in quantityList of objectEvent data', () => {
+      const data = fs.readFileSync(`${process.cwd()}/${credentials[1].dataPath}`, 'utf8');
+      const jsonData = JSON.parse(data);
+      expect(jsonData.quantityList[1].volume).toEqual(expect.any(String));
+    });
+
+    it('should not exist the quantity in transformationEvent data', () => {
+      const data = fs.readFileSync(`${process.cwd()}/${credentials[2].dataPath}`, 'utf8');
+      const jsonData = JSON.parse(data);
+      expect(jsonData.outputQuantityList[0]).not.toHaveProperty('quantity');
+    });
```

Comment:
What is the purpose of these cases?

Reply:
In this case, I want to show why the final report is FAIL.